sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -206,23 +207,16 @@ internal string QueryAnswerRaw
                 if (string.IsNullOrEmpty(value))
                 {
                     QueryAnswer = null;
+                    QueryAnswerCount = null;
                 }
                 else
                 {
                     string[] queryAnswerParts = value.Split('|');
                     QueryAnswer = new QueryAnswer(queryAnswerParts[0]);
 
-                    if (queryAnswerParts.Length == 1)
-                    {
-                        if (QueryAnswer.Value == Models.QueryAnswer.Extractive)
-                        {
-                            // Set answer count to the default value.
-                            QueryAnswerCount = 1;
-                        }
-                    }
-                    else
+                    if (queryAnswerParts.Length > 1)
                     {
-                        string countString = queryAnswerParts[1].Length > 6 ? queryAnswerParts[1].Substring(6) : "1";
+                        string countString = queryAnswerParts[1].Length > 6 ? queryAnswerParts[1].Substring(6) : null;
```

Comment:
Can you add a comment explaining where this magic number 6 comes from?  Is it the "count-"?  Should we verify that's the actual text in case a future service version changes this format?

Reply:
I restructured the code to not use magic numbers.